### PR TITLE
algorithm: binary insertion sort

### PR DESCRIPTION
**Pull request by syedjafer**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$1181}`.

### CODE REVIEW

File: Sorts/BinaryInsertionSort.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    return binarySearch(array, key, mid + 1, end)
+  } else if (array[mid] > key) {
+    return binarySearch(array, key, start, mid - 1)
+  } else {
+    return mid
+  }
+}
+
+/**
+   * Binary Insertion Sort
```
Comment (appgurueu) [inline Sorts/BinaryInsertionSort.js:46]: Please unindent this.

File: Sorts/BinaryInsertionSort.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+   *
+   * @param {Array} list List to be sorted.
+   * @return {Array} The sorted list.
+   */
+export function binaryInsertionSort (array) {
+  const totalLength = array.length
+  for (let itr = 1; itr < totalLength; itr += 1) {
```
Comment (appgurueu) [inline Sorts/BinaryInsertionSort.js:53]: `itr`? Please call this `i` (or `idx` if you want a longer name).

Comment (syedjafer) [reply Sorts/BinaryInsertionSort.js]: Resolved. Changed to **i**

**Review by appgurueu — DISMISSED** [review]

File: Maths/CollatzSequence.js [hunk 3]
```diff
@@ -26,5 +26,5 @@ export function collatz (n) {
     steps.push(n)
   }
 
-  return { result: n, steps: steps }
+  return { result: n, steps }
```
Comment (raklaptudirm) [inline Maths/CollatzSequence.js:29]: This change is not related to this pr, so undo this.

**Review by raklaptudirm — APPROVED** [review]

File: Maths/CollatzSequence.js (hunk 3 above)
Comment (syedjafer) [reply Maths/CollatzSequence.js]: resolved

**Comment by syedjafer** [comment]
I have resolved all the comments. @appgurueu @raklaptudirm  can u pls confirm this PR closure. 
Thanks

**Review by raklaptudirm — APPROVED** [review]

**Review by appgurueu — APPROVED** [review]
